src/ImpactEvaluator.sol

```diff
@@ -84,45 +82,48 @@ contract ImpactEvaluator is AccessControl {
             addresses.length == scores.length,
             "Addresses and scores length mismatch"
         );
-        require(roundIndex < currentRoundIndex, "Round not finished");
-
-        Round storage round = openRounds[roundIndex];
-        require(round.exists, "Open round does not exist");
+        require(roundIndex < currentRoundIndex(), "Round not finished");
 
+        Round storage round = getOpenRound(roundIndex);
         uint sumOfScores = validateScores(scores, round.totalScores);
         reward(addresses, scores);
         round.totalScores += sumOfScores;
 
         if (round.totalScores == MAX_SCORE) {
-            deleteRound(roundIndex);
+            deleteOpenRound(roundIndex);
         }
     }
 
-    function deleteRound(uint roundIndex) private {
-        delete openRounds[roundIndex];
+    function getOpenRound(
+        uint roundIndex
+    ) private view returns (Round storage) {
+        for (uint i = 0; i < openRounds.length; i++) {
+            if (openRounds[i].index == roundIndex) {
+                return openRounds[i];
+            }
+        }
+        revert("Open round does not exist");
+    }
 
-        // Find index inside `openRoundIndexes` and remove it while shrinking
+    function deleteOpenRound(uint roundIndex) private {
+        // Find index inside `openRounds` and remove it while shrinking
```

Comment:
Hard to test because tests pass if I don't delete the open round. I think it's very necessary to shrink the array, because otherwise eventually it will be too big/too expensive for the contract to run.

Good idea to optimize like this, I propose to reevaluate this later on, as it also complicates things